rclpy/rclpy/node.py

```diff
@@ -657,6 +657,65 @@ def get_parameters(self, names: List[str]) -> List[Parameter]:
             raise TypeError('All names must be instances of type str')
         return [self.get_parameter(name) for name in names]
 
+    def list_parameters(self, prefixes: Optional[List[str]] = None, depth: int = 0) -> List[str]:
```

Comment:
I know we aren't totally consistent with this in `rclpy`, but I do think that we should in general aim to make these APIs work like the `rclcpp` ones.  In this case, I think that means that we should return a message of type `rcl_interfaces/msg/ListParametersResult`, rather than just a list of strings.